### ISSUE TITLE
--json should return strictly json format outputs..

### ISSUE THREAD

**Issue by mkliu** [Azure/azure-xplat-cli#2049]
I guess the reason why we have this is to enable automation. However there are many commands which doesn't return json formatted strings and could completely break automation flow. 

For example, when i create a new resource group, it returns a weird format output with a bunch of strings. I don't even need the output to make sense, but I need it to be properly formatted

**Comment by mkliu** [comment]
same issue with those which does return json, the format isn't right. For example when i do azure group deploy --json, I want to look at the output variables from my deployment, but the outputted string is really hard to parse

**Comment by yugangw-msft** [comment]
@mkliu
yes "--json" is for automation and not formatted to let human read. Any reason you can't use the non json option?

**Comment by mkliu** [comment]
@yugangw-msft I want to use it for automation. But the returned value isn't really json. For example, creating a new ARM deploy returns something like this:

Am I supposed to break down each new line and analyze myself?


```
"test2"
"Succeeded"
"2015-08-21T10:34:55.9781606Z"
"Incremental"
{
  "environment": {
    "type": "String",
    "value": "dev"
  },
  "sqlServerName": {
    "type": "String",
    "value": "newappdev"
  },
  "sqlServerLocation": {
    "type": "String",
    "value": "East US"
  },
  ...
}
```

**Comment by sedouard** [comment]
+1000 To this one. Why is it that random commands decide not to output proper JSON with the `--json` flag? This breaks automation. If output can be tabular, there should be a compatible JSON adapted view as well, right?

**Comment by sedouard** [comment]
example: `azure ad app create`

**Comment by amarzavery** [comment]
@sedouard - The issue of incorrect json output for ad app create has been fixed.

**Comment by amarzavery** [comment]
@vivsriaus - Has this been fixed? If yes then please close the issue.

**Comment by vivsriaus** [comment]
@amarzavery Yes, the --json format issue we had with arm commands was fixed a while ago.
